tensorflow/python/ops/array_ops.py

```diff
@@ -3532,7 +3546,76 @@ def _get_paddings_constant(paddings):
     return [_get_paddings_constant(x) for x in paddings]
   else:
     return paddings
+  
+def _wrap_pad(tensor, paddings, name=None):
+  """Helper to create wrap pad option with ops already built.
 
+  See `pad` for more info on how this function works.
+  """
+  # use constant padding to create a tensor with final shape and original values.
+  padded = gen_array_ops.pad_v2(input=tensor, paddings=paddings, name=name)
+  # store the begin and end of original slice area
+  slice_area_to_copy = [
+    _BaseSlice(dim - paddings.numpy()[index][1] - tensor.shape[index], 
+          dim - paddings.numpy()[index][1]) 
+          for index, dim in enumerate(padded.shape)]
```

Comment:
Same here, you can't iterate over .shape.  I'm not sure if you can iterate over `tf.shape(...)` either - you would need to check.  Try calling this from within a `@tf.function`.

Reply:
You can't iterate over tf.shape(...) within a `@tf.function` but you can iterate over tf.range(tf.shape(...)[0]) and access the relevant elements that way.